packages/extension/src/components/pages/NFTs/NFTs.tsx

```diff
@@ -0,0 +1,62 @@
+import { FC, useContext, useEffect, useState } from 'react';
+
+import { AccountNFTokenResponse } from '@gemwallet/constants';
+
+import { LedgerContext } from '../../../contexts';
+import { NFTListing } from '../../organisms';
+import { PageWithHeader } from '../../templates';
+
+export const MAX_FETCHED_NFTS = 20;
+
+const initalState = {
+  account_nfts: [],
+  marker: null,
+  loading: false
+};
+
+interface NFTsProps extends AccountNFTokenResponse {
+  loading: boolean;
+}
+
+export const NFTs: FC = () => {
+  const { getNFTs } = useContext(LedgerContext);
+
+  const [NFTs, setNFTs] = useState<NFTsProps>(initalState);
+
+  const fetchNFTs = async () => {
+    try {
+      const payload = {
+        limit: MAX_FETCHED_NFTS,
+        marker: NFTs.marker ?? undefined
+      };
+
+      setNFTs({ ...NFTs, loading: true });
+
+      const response = await getNFTs(payload);
+
+      setNFTs({
+        marker: response.marker,
+        account_nfts: NFTs.account_nfts.concat(response.account_nfts),
```

Comment:
Why do we need to concat? Isn't the initial state an empty array?

Would this be simpler?

```jsx
const { marker, account_nfts } = await getNFTs(payload);

setNFTs({
        marker,
        account_nfts,
        loading: false
});
```

Reply:
It's because everytime we fetch more NFTs using the marker, we concatenate them with the previous NFTs that have already been fetched

```
<NFTListing
        {...{
          ...NFTs,
          onLoadMoreClick: fetchNFTs
        }}
      />
```

```
  const fetchNFTs = async () => {
    try {
      const payload = {
        limit: MAX_FETCHED_NFTS,
        marker: NFTs.marker ?? undefined
      };

      setNFTs({ ...NFTs, loading: true });

      const response = await getNFTs(payload);

      setNFTs({
        marker: response.marker, // --> set the marker here
        account_nfts: NFTs.account_nfts.concat(response.account_nfts),
        loading: false
      });
    } catch (error) {
      setNFTs(initalState);
    }
```